packages/core/src/model.js

```diff
@@ -2481,11 +2475,15 @@ ${associationOwner._getAssociationDebugList()}`);
 
     this._injectScope(options);
 
-    if (!options || !(options.where || options.truncate)) {
-      throw new Error('Missing where or truncate attribute in the options parameter of model.destroy.');
+    if (options && 'truncate' in options) {
+      throw new Error('Model#destroy does not support the truncate option. Use Model#truncate instead.');
     }
 
-    if (!options.truncate && !isPlainObject(options.where) && !Array.isArray(options.where) && !(options.where instanceof BaseSqlExpression)) {
+    if (!options?.where) {
+      throw new Error('Missing where attribute in the options parameter of model.destroy.');
+    }
+
+    if (!isPlainObject(options.where) && !Array.isArray(options.where) && !(options.where instanceof BaseSqlExpression)) {
```

Comment:
This condition isn't necessary anymore, it will be handled by the query generator

Reply:
I've removed the conditions now